src/util.cpp

```diff
@@ -37,13 +38,20 @@ std::string toHex(const std::vector<uint8_t> &data) {
     return result.str();
 }
 
-std::vector<uint8_t> randomBytes(size_t length) {
-    std::vector<uint8_t> buffer(length);
-    if (1 != RAND_bytes(buffer.data(), buffer.size())) {
-        // TODO: throw?
+std::vector<uint8_t> fromHex(const std::string &hexData) {
+    std::vector <uint8_t> binary;
```

Comment:
we already know that we will need hexData.size()/2 bytes here and should make sure that hexData has even length. Might lead to strange behavior otherwise.

Reply:
fixed